RecoParticleFlow/PFClusterProducer/plugins/PFClusterFromHGCalTrackster.cc

```diff
@@ -47,17 +47,11 @@ void PFClusterFromHGCalMultiCluster::buildClusters(const edm::Handle<reco::PFRec
     double energy = 0.0, highest_energy = 0.0;
     output.emplace_back();
     reco::PFCluster& back = output.back();
-    const auto& hitsAndFractions_mcl = mcl.hitsAndFractions();
 
     std::vector<std::pair<DetId, float> > hitsAndFractions;
-    hitsAndFractions.insert(hitsAndFractions.end(), hitsAndFractions_mcl.begin(), hitsAndFractions_mcl.end());
-
-    // Use the H&F of the clusters inside the multicluster if the latter's H&F are not stored
-    if (hitsAndFractions.empty()) {
-      for (const auto& cl : mcl) {
-        const auto& hAndF_temp = cl->hitsAndFractions();
-        hitsAndFractions.insert(hitsAndFractions.end(), hAndF_temp.begin(), hAndF_temp.end());
-      }
+    for (const auto lcId : tst.vertices()) {
+      const auto& hAndF_temp = clusters[lcId].hitsAndFractions();
+      hitsAndFractions.insert(hitsAndFractions.end(), hAndF_temp.begin(), hAndF_temp.end());
```

Comment:
Didn't you forget to apply the `multiplicity` here? I believe this is a bug.
See [here](https://github.com/cms-sw/cmssw/blob/master/RecoHGCal/TICL/plugins/MultiClustersFromTrackstersProducer.cc#L73-L96), for example.

Reply:
Right, done.